### ISSUE TITLE
[Migration] Introduction of the System Template

### ISSUE THREAD

**Issue by darshanasbg** [wso2/product-is#21304]
**Describe the issue:**
With the introduction of the System Templates with https://github.com/wso2/product-is/issues/20996, there is a minor behavior change in the API.

Quoting the design document,
`There is a minimal impact where when we try to delete a system template it won’t get deleted but instead it will get reverted back to the default state.`

Having the ability to delete system default template, leads to failures in the scenarios that needs these templates, so not giving the ability to delete these templates considered as a bug fix, that avoid having the user to lead such errors in the runtime..

However, this change must be noted in the migration guide.

**How to reproduce:**
Delete a email templates that is available in the default pack (in a fresh tenant). In earlier version delete operation completely delete the template. If the same template try to retrieve after the delete, it is returning 404..

But with 7.1.0 deleting the template, doesn't lead to delete the resource completely, rather its reset the content to default (if there are any changes has been made).. And if the same template try to retrieve after the delete request, its returning the template with default content (rather returning a 404).

**Expected behavior:**
N/A

**Comment by AnuradhaSK** [comment]
Due to the introduction of system email templates, a behavior change can be ocurred in email sending for account locking scenario due to the code logic as follows:

https://github.com/wso2-extensions/identity-event-handler-account-lock/blob/71a72cab5555445f5a15ac49165a8990f1850219/components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java#L733-L747

This code was introduced with https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/58

```
 String emailTemplateTypeAccUnlocked = AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_UNLOCKED;
                if (notificationInternallyManage) {
                    if (isAdminInitiated) {
                        if (AccountUtil
                                .isTemplateExists(AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_UNLOCKED_ADMIN_TRIGGERED,
                                        tenantDomain)) {
                            emailTemplateTypeAccUnlocked =
                                    AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_UNLOCKED_ADMIN_TRIGGERED;
                        }
                    } else {
                        if (AccountUtil.isTemplateExists(AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_UNLOCKED_TIME_BASED,
                                tenantDomain)) {
                            emailTemplateTypeAccUnlocked = AccountConstants.EMAIL_TEMPLATE_TYPE_ACC_UNLOCKED_TIME_BASED;
                        }
                    }
```

Possible behavior changes:
1. If the earlier IS tenants didn't have `accountUnlockTimeBased`/ `accountUnlockAdmin` email templates, the endusers have recieved `accountUnlock` template for both cases of unlocking user account by unlock time passing and admin unlocking the user cases. With new system email template support `accountUnlockTimeBased` and `accountUnlockAdmin` will be available for that tenant and email triggering will pick the specific scenario based unlock template. If the `accountUnlock` has been customized, there is a behavior change
2.

**Comment by Thumimku** [comment]
I am removing my assignment from this issue as I am no longer working on it. However, I have a proposed solution.

The ```isTemplateExists``` method currently takes two parameters: the email template name and the tenant domain. I suggest adding a third parameter—a boolean variable—to indicate whether the existence of the template should also be checked in the ```defaultTemplateManager```.

CC:
@nilasini @darshanasbg

**Comment by nilasini** [comment]
@Thumimku yes since this is a behavioural change we have get the issue mentioned in [1] fixed by using a configuration. For the issue mentioned in the [2] you need to update the migration doc.

[1] https://github.com/wso2/product-is/issues/21304#issuecomment-2402110221
[2] https://github.com/wso2/product-is/issues/21304#issue-2575505108

**Comment by Thumimku** [comment]
@AnuradhaSK ,
CC: @darshanasbg 
As far as I know, here is our ideal practice:  

Before version 7.1.0, we used registry-based email templates, which were automatically inserted into the registry when a tenant was created (in addition to any manual insertions). When introducing a new email template, we required an email migrator to ensure these templates were added in future versions.  From IS 7.0.0, we don't have such practise and we added two email templates OrganizationUserInvitation, ImpersonationEmailNotification which will be added via email migrator. 

The issue you're referring to stems from the **same flow having two different email templates**, which was already present before version 7.0.0. Therefore, this problem existed regardless of the introduction of the system template. 

I agree that we need to address this issue. I will create a separate ticket to track it, as the problem is independent of the newly introduced feature.

**Comment by Thumimku** [comment]
As per above comment the issue mentioned by @AnuradhaSK is orthogonal to the feature we untorduced, and the issue tracked as separate issue hence I am closing this issue.

CC: @darshanasbg @nilasini @AnuradhaSK @DMHP
